api/v1beta1/azuremachinetemplate_webhook_test.go

```diff
@@ -128,15 +128,109 @@ func TestAzureMachineTemplate_ValidateCreate(t *testing.T) {
 			wantErr: true,
 		},
 		{
-			name:            "azuremachinetemplate with RoleAssignmentName",
-			machineTemplate: createAzureMachineTemplateFromMachine(createMachineWithRoleAssignmentName()),
+			name:            "azuremachinetemplate with SystemAssignedIdentityRoleName",
+			machineTemplate: createAzureMachineTemplateFromMachine(createMachineWithSystemAssignedIdentityRoleName()),
 			wantErr:         true,
 		},
+		{
+			name:            "azuremachinetemplate without SystemAssignedIdentityRoleName",
+			machineTemplate: createAzureMachineTemplateFromMachine(createMachineWithoutSystemAssignedIdentityRoleName()),
+			wantErr:         false,
+		},
 		{
 			name:            "azuremachinetemplate without RoleAssignmentName",
 			machineTemplate: createAzureMachineTemplateFromMachine(createMachineWithoutRoleAssignmentName()),
 			wantErr:         false,
 		},
+		{
```

Comment:
nit: Can you move this test above the one without RoleAssignmentName just for consistency?

Reply:
That makes sense! Made the change